trino/trino.go

```diff
@@ -678,21 +698,37 @@ func (st *driverStmt) QueryContext(ctx context.Context, args []driver.NamedValue
 		stmt:    st,
 		queryID: sr.ID,
 		nextURI: sr.NextURI,
+		statsCh: st.statsCh,
 	}
 	if err = rows.fetch(false); err != nil {
 		return nil, err
 	}
+	close(st.doneCh)
 	return rows, nil
 }
 
 func (st *driverStmt) exec(ctx context.Context, args []driver.NamedValue) (*stmtResponse, error) {
+	st.statsCh = make(chan QueryProgressInfo)
+	st.doneCh = make(chan struct{})
 	query := st.query
 	var hs http.Header
 
 	if len(args) > 0 {
 		hs = make(http.Header)
 		var ss []string
 		for _, arg := range args {
+			if arg.Name == trinoProgressCallbackParam {
+				callback := arg.Value.(ProgressUpdater)
```

Comment:
Do you need `callback` and `period` variables, or you can just inline those and assign directly to `st.conn.callback`st.conn.callbackPeriod.Period`?
 I recall there was some go weirdness that you sometimes needed to copy iteration variable into new one - but I do not think this is the case.